internal/experiment/generation/quantity.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2021 GramLabs, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package generation
+
+import (
+	"k8s.io/apimachinery/pkg/api/resource"
+)
+
+// AsScaledInt scales the quantity to the appropriate scale, honoring the base
+// as determined by the format of the supplied quantity. For example,
+// AsScaledInt(NewQuantity(1, BinarySI), Milli) returns 1024.
+func AsScaledInt(q resource.Quantity, scale resource.Scale) int32 {
+	// ScaledValue works fine for base 10
+	if q.Format != resource.BinarySI {
+		return int32(q.ScaledValue(scale))
+	}
+
+	v := q.Value()
+	if scale > 0 {
+		for e := int(scale) / 3; e > 0; e-- {
+			v /= 1024
+		}
+	} else {
+		for e := int(scale) / 3; e < 0; e++ {
+			v *= 1024
+		}
+	}
+	return int32(v)
+}
+
+type suffixList []string
+
+func (l suffixList) lookup(scale resource.Scale) string {
```

Comment:
Can we add a comment to this?

Im pretty sure it's to normalize the value of scale so we can use this as the index for the suffixLists below, but not fully understanding what resource.Scale ( https://pkg.go.dev/k8s.io/apimachinery/pkg/api/resource#Scale ) was ahead of time took me a bit to work through this.